xonsh/pyghooks.py

```diff
@@ -1388,7 +1388,10 @@ def color_file(file_path: str, mode: int) -> (Color, str):
         except FileNotFoundError:
             color_key = "or"
     elif stat.S_ISREG(mode):
-        if stat.S_IMODE(mode) & (stat.S_IXUSR + stat.S_IXGRP + stat.S_IXOTH):
+        cap = os.listxattr(file_path)
+        if cap and "security.capability" in cap:    # protect None return on some OS?
+            color_key = "ca"
+        elif stat.S_IMODE(mode) & (stat.S_IXUSR + stat.S_IXGRP + stat.S_IXOTH):
             color_key = "ex"
         elif (
```

Comment:
`su` and `sg` should have both higher precedence than `ca` and `ex`, see https://github.com/coreutils/coreutils/blob/master/src/ls.c#L4922.

Reply:
Thank you for finding a reference implementation!  This allowed me to correct actually several errors, the precedence issue you mention plus a confusion of 'or' and 'mi'.  Plus that code shows how to implement 'ln=target', which I will include.

Reply:
I had mentioned that reference in #3608 several times already. :-)

Thank you for your work! I'll look at this PR again once you have target and ca fixed.